File: app/models/qdm/substance_nucleic_acid.rb
module QDM
  # app/models/qdm/substance_nucleic_acid.rb
  class SubstanceNucleicAcid < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :sequenceType
    field :numberOfSubunits
    field :areaOfHybridisation
    field :oligoNucleotideType
    field :subunit
    field :qdmVersion, type: String, default: '4.0.0'
  end
end

File: app/assets/javascripts/SubstanceNucleicAcid.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const SubstanceNucleicAcidSchema = DataElementSchema({
  sequenceType: Any,
  numberOfSubunits: Any,
  areaOfHybridisation: Any,
  oligoNucleotideType: Any,
  subunit: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'SubstanceNucleicAcid' },

});

module.exports.SubstanceNucleicAcidSchema = SubstanceNucleicAcidSchema;
module.exports.SubstanceNucleicAcid = mongoose.model('SubstanceNucleicAcid', SubstanceNucleicAcidSchema);
